sys-cluster/charliecloud/charliecloud-0.9.10.ebuild

```diff
@@ -0,0 +1,67 @@
+# Copyright 1999-2019 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+# User namespaces don't play well with the sandbox.
+RESTRICT="test"
```

Comment:
Could you move that below IUSE? It's surprising to find it on the top of file.

Also, can you try disabling sandbox via `local SANDBOX_ON=0`?

Reply:
Of course!
Sadly, even disabling the sandbox does not help for this container runtime: The tests check consistency of namespaces outside and inside a test container (e.g. mount namespace, effects of `setgid`), and this is confused by the special privileges and namespaces that seem to be used by portage itself. 
Should I remove the test section completely, or do you have another suggestion?

Reply:
If the test code is otherwise fine, let's leave it as-is for now. Maybe we'll have some good workaround in the future. Possibly report this upstream and suggest they could make the tests less fragile.

Reply:
I checked a bit deeper through all failing tests (continuing after the permission / namespace stuff) and have bad news: The smallest sufficient set of the test suite needs to download external container images during testing (which appears to fail already in the namespaced portage environment, and of course these downloaded things may change anytime...). 
These containers themselves also need web access to test package installation inside the containers, and run tests there. 
For better coverage, the testing code hints that one should also have Docker installed and running and then the tests automagically pull images (so the tests may change system state outside of the build process!). . 
All that in addition to the permission issues... 

While I think this test suite is good in the sense that it has really large coverage over the runtime (ideal for developers and regression tests e.g. using travis - the tests-code is a manifold larger than the actual program!), it does not really fit into a build process and build environment. 

The package also supports running the tests in user space after installation ( https://hpc.github.io/charliecloud/test.html ) but I don't see a clean way to do this in portage. 

So it would probably be more useful to completely drop the tests from the ebuild. What do you think?

Reply:
Yeah, that sounds like a show stopper. However, it might be possible to fetch those files via `SRC_URI` and place them in some location the test runner expects. I'm not going to ask you to fight that though. If you think the current code is of little value to anyone who might try to make tests work in the future, feel free to drop it. But generally keeping it doesn't hurt if it has some value (read: if it's not entirely wrong).

Reply:
> However, it might be possible to fetch those files via `SRC_URI` and place them in some location the test runner expects.

That would only fix part of the issue - the suite also runs `apt-get` and `yum install` in those containers... 
Since the part of the `test` code which I have in the ebuild is easily reproducible with the documentation, I dropped it just now.